frontend/src/components/ResultsList.js

```diff
@@ -42,15 +42,19 @@ const sampleJSON = {
         },
 }
 
-const ResultsList = () => (
+const ResultsList = ({difficulty}) => (
     <Fragment>
         <div className="results-container">
-            {Object.keys(sampleJSON).map(app => 
-                <Result 
-                    name={sampleJSON[app].name} 
-                    description={sampleJSON[app].description} 
-                    link={sampleJSON[app].link} 
-                />
+            {Object.keys(appIdeas).map(function (app) { 
```

Comment:
Think about it, if `appIdeas` is an array of objects you will only need to call map on the array and that's all. However, as an object, you first called Object.keys which returns an array before you then map the array. So, you can save that extra computation and improve app performance by reducing the time complexity to O(n) rather than the O(n2) that it currently is. The point essentially is, one loop is less expensive to compute than two loops.